source/opt/def_use_manager.h

```diff
@@ -53,13 +54,41 @@ inline bool operator<(const Use& lhs, const Use& rhs) {
   return lhs.operand_index < rhs.operand_index;
 }
 
-using UseList = std::list<Use>;
+using UserEntry = std::pair<ir::Instruction*, ir::Instruction*>;
```

Comment:
Which is the use and which is the user?  Must say so.
I'm going to assume the .first entry is the thing being used.  The .second entry is the thing using it.
Are they allowed to be nullptr?  And what does that mean?

Reply:
I'll add comments.